Update docs for split stdout / stderr Pod logging (#48405)
kubernetes/website: content/en/docs/concepts/cluster-administration/logging.md

kubectl logs counter -c count
```

> Since v1.32, you can request a specific log stream of a container in a running Pod.
> This functionality requires the
> `PodLogsQuerySplitStreams` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) to be enabled.

Reviewer: ```suggestion

### Container log streams

{{< feature-state feature_gate_name="PodLogsQuerySplitStreams" >}}

As an alpha feature, the kubelet can split out the logs from the two standard streams produced
by a container: [standard output](https://en.wikipedia.org/wiki/Standard_streams#Standard_output_(stdout))
and [standard error](https://en.wikipedia.org/wiki/Standard_streams#Standard_error_(stderr)).
To use this behavior, you must enable the `PodLogsQuerySplitStreams`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
```

> Unfortunately, only the server side changes have been merged so far; the [corresponding kubectl changes](https://github.com/kubernetes/kubernetes/pull/128159) are still under review.
> In the meantime, you can still use this feature by directly accessing the API.

Reviewer: We prefer to write documentation that is more [timeless](https://developers.google.com/style/timeless-documentation). Try this:
```suggestion
With that feature gate enabled, Kubernetes {{< skew currentVersion >}} allows access to these
log streams directly via the Pod API. You can fetch a specific stream by specifying the stream name (either `Stdout` or `Stderr`),
using the `stream` query string. You must have access to read the `log` subresource of that Pod.
```

To demonstrate this feature, you can create a Pod that periodically writes text to both the standard output and error stream.

{{% code_sample file="debug/counter-pod-err.yaml" %}}

To run this pod, use the following command:

```shell
kubectl apply -f https://k8s.io/examples/debug/counter-pod-err.yaml
```

To fetch only the stderr log stream, you can run:

```shell
kubectl get --raw "/api/v1/namespaces/default/pods/counter-err/log?stream=Stderr"
```

> Once kubectl is updated, you will be able to easily retrieve a specific log stream using:
> 
> ```shell
> kubectl logs --stream=stderr counter-err
> ```

Reviewer: We [avoid statements about the future](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future) in docs, so:
```suggestion
```
